lib/charms/bind/v0/dns_record.py

```diff
@@ -347,40 +350,71 @@ def to_relation_data(self, model: ops.Model, relation: ops.Relation) -> Dict[str
         return dumped_data
 
     @classmethod
-    def from_relation(cls, model: ops.Model, relation: ops.Relation) -> "DNSRecordRequirerData":
+    def from_relation(
+        cls, model: ops.Model, relation: ops.Relation
+    ) -> Tuple["DNSRecordRequirerData", "DNSRecordProviderData"]:
         """Initialize a new instance of the DNSRecordRequirerData class from the relation data.
 
         Args:
             model: the Juju model.
             relation: the relation.
 
         Returns:
-            A DNSRecordRequirerData instance.
+            the DNSRecordRequirerData instance.
```

Comment:
I think it might help explaining the tuple: (Valid DNSRecordRequirerData, Invalid DNSRecordProviderData) :D